fix: add the migration guide and nunjucks depreciation notes (#1253)
asyncapi/generator: .changeset/depreciate-nunjucks.md

---
"@asyncapi/generator": major
---

> Migration guide for nunjucks render engine to react render engine

Reviewer: ```suggestion
Migrating from Nunjucks render engine to React render engine
```
